packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,263 @@
+      placeholder={placeholder}
+      className={cx(
+        selectStyles,
+        css({ minWidth: `calc(${longestLabel}ch + ${spacing[6]}px)` })
+      )}
+      onChange={onSelect}
+      value={value}
+    >
+      {options.map((option) => (
+        <Option key={option} value={option}>
+          {option}
+        </Option>
+      ))}
+    </Select>
+  );
+};
+
+function useSearchFilter(): [React.ReactElement, string] {
+  const [search, setSearch] = useState('');
+  const searchControls = useMemo(() => {
+    return (
+      <TextInput
+        className={searchInputStyles}
+        aria-label="Search"
+        type="search"
+        placeholder="Search"
+        value={search}
+        onChange={(e) => {
+          setSearch(e.target.value);
+        }}
+        spellCheck={false}
+      />
+    );
+  }, [search]);
+
+  return [searchControls, search];
+}
+
+function useSelectFilter(items: Item[]): [React.ReactElement, SelectState] {
+  const [selectState, setSelectState] = useState<SelectState>({});
+  const [collections, setCollections] = useState<string[]>([]);
+
+  const databases = _.uniq(_.map(items, 'database'));
+
+  const selectDatabase = useMemo(() => {
+    return (database: string): void => {
+      setSelectState({
+        database: database ?? undefined,
+      });
+
+      const collections = _.uniq(
+        _.map(
+          _.filter(items, (x) => x.database === database),
+          'collection'
+        )
+      );
+      setCollections(collections);
+    };
+  }, [items]);
+
+  const selectCollection = useMemo(() => {
+    return (collection: string): void => {
+      setSelectState({
+        ...selectState,
+        collection: collection ?? undefined,
+      });
+    };
+  }, [selectState]);
+
+  const selectControls = useMemo(() => {
+    return (
+      <div className={selectContainer}>
+        <FilterSelect
+          options={databases}
+          placeholder={'All databases'}
+          onSelect={selectDatabase}
+          value={selectState.database}
+        />
+        <FilterSelect
+          options={collections}
+          placeholder={'All collections'}
+          onSelect={selectCollection}
+          value={selectState.collection}
+        />
+      </div>
+    );
+  }, [databases, selectDatabase, selectState, collections, selectCollection]);
+
+  return [selectControls, selectState];
+}
+
+function filterItemByConditions(item: Item, conditions: SelectState): boolean {
+  if (Object.keys(conditions).length === 0) {
+    return true;
+  }
+  let shouldReturnItem = true;
+  for (const key in conditions) {
+    if (
+      item[key as keyof SelectState] !== conditions[key as keyof SelectState]
+    ) {
+      shouldReturnItem = false;
+    }
+  }
+  return shouldReturnItem;
+}
+
+function filterByText(items: Item[], text: string): Item[] {
+  if (!text || text.length === 1) {
+    return items;
+  }
+  const fuse = new Fuse<Item>(items, {
+    findAllMatches: true,
+    shouldSort: true,
```

Comment:
As I mentioned in another place, I think we should try to keep the concepts separate, otherwise it's hard to understand at a glance what is responsible for what here (especially as there is no indication in the hook names that they will also do sorting)